src/dataframe/dataframe.jl

```diff
@@ -1317,3 +1317,77 @@ end
 function permutecols!(df::DataFrame, p::AbstractVector{Symbol})
     permutecols!(df, index(df)[p])
 end
+
+##############################################################################
+##
+## Expand & Complete dataframes
+##
+##############################################################################
+
+function _expandhelper(col, vect, stride, offset)
+    for i in offset:stride:length(vect)
+        push!(col, vect[i])
+    end
+end
+
+function expand(df::AbstractDataFrame, indexcols)
+    # Check to make sure the symbols in indexcols are in the df, and check for duplicate rows in the input df
+    allunique(indexcols) || throw(ArgumentError("Elements of $indexcols must be unique"))
+    sum(DataFrames.nonunique(df[:,indexcols]))>0 && @warn "duplicate rows in input; expand will only return unique combinations"
```

Comment:
```suggestion
    length(nonunique(df, indexcols))>0 && @warn "duplicate rows in input; expand will only return unique combinations"
```

Reply:
I think using `sum(nonunique(df, indexcols))>0` is the correct test here as `length(nonunique(df, indexcols))>0` always returns true - the returned value of `nonunique(df, indexcols)` is an array of Bools that will be >0.

Reply:
Ah right (I mistakenly thought we return the actual indices).